windows/text.cpp

```diff
@@ -66,20 +87,39 @@ int uiWindowsWindowTextWidth(HWND hwnd)
 		ReleaseDC(hwnd, dc);
 		goto noTextOrError;
 	}
-	if (GetTextExtentPoint32W(dc, text, len, &size) == 0) {
-		logLastError(L"error getting text extent point");
-		// continue anyway, assuming size is 0
-		size.cx = 0;
-		size.cy = 0;
+
+	// calculate width of each line
+	while(start != text + len) {
+		while(*start == '\n' && start != text + len) {
+			start++;
+		}
+		if(start == text + len) {
+			break;
+		}
+		end = start + 1;
+		while(*end != '\n' && end != text + len){
+			end++;
+		}
+		if (GetTextExtentPoint32W(dc, start, end - start, &size) == 0) {
+			logLastError(L"error getting text extent point");
+			// continue anyway, assuming size is 0
+			size.cx = 0;
+			size.cy = 0;
+		}
+		if(size.cx > maxWidth) {
+			maxWidth = size.cx;
+		}
```

Comment:
```suggestion
		if (GetTextExtentPoint32W(dc, start, end - start, &size) == 0)
			logLastError(L"error getting text extent point");
		else
			if (size.cx > maxWidth)
				maxWidth = size.cx;
```
Looks clearer to me and removes a bunch of lines this way.